src/vic3_world/world/vic3_world_builder_tests.cpp

```diff
@@ -0,0 +1,73 @@
+#include "external/commonItems/external/googletest/googlemock/include/gmock/gmock-matchers.h"
+#include "external/commonItems/external/googletest/googletest/include/gtest/gtest.h"
+#include "src/vic3_world/world/vic3_world.h"
+namespace vic3
+{
+
+TEST(Vic3WorldWorldVic3WorldBuilder, NullBuilderOutputsNull)
+{
+   const auto nullWorld = WorldBuilder::CreateNullWorld().Build();
+
+   EXPECT_TRUE(nullWorld.GetCountries().empty());
+   EXPECT_TRUE(nullWorld.GetProvinceDefinitions().GetProvinceDefinitions().empty());
+}
+
+TEST(Vic3WorldWorldVic3WorldBuilder, AddProvincesWorks)
+{
+   const auto nullWorld = WorldBuilder::CreateNullWorld().AddProvinces({{"prov1"}, {"prov2"}}).Build();
+
+   EXPECT_EQ(nullWorld.GetProvinceDefinitions().GetProvinceDefinitions().at(0), "prov1");
+   EXPECT_EQ(nullWorld.GetProvinceDefinitions().GetProvinceDefinitions().at(1), "prov2");
+}
+
+TEST(Vic3WorldWorldVic3WorldBuilder, AddStatesWorks)
+{
+   const auto nullWorld =
+       WorldBuilder::CreateNullWorld().AddStates({{1, State({.owner_number = 1, .provinces = {1, 2, 3}})}}).Build();
+
+   EXPECT_EQ(nullWorld.GetStates().at(1), State({.owner_number = 1, .provinces = {1, 2, 3}}));
+}
+
+TEST(Vic3WorldWorldVic3WorldBuilder, AddTestStatesWorks)
+{
+   const auto nullWorld = WorldBuilder::CreateNullWorld().AddTestStates({{1, 2, 3}, {4, 5}}).Build();
+
+   EXPECT_EQ(nullWorld.GetStates().at(1), State({.owner_number = 1, .provinces = {1, 2, 3}}));
+   EXPECT_EQ(nullWorld.GetStates().at(2), State({.owner_number = 2, .provinces = {4, 5}}));
+}
+
+TEST(Vic3WorldWorldVic3WorldBuilder, AddStateRegionsWorks)
+{
+   const auto nullWorld = WorldBuilder::CreateNullWorld()
+                              .AddStateRegions({
+                                  {"REGION_ONE", vic3::StateRegion({{"0x000002", "port"}}, {})},
```

Comment:
I wasn't always consistent about this at first, but Vic3 province numbers don't have the leading 0. It'd just be x000002. I haven't indicated the other instances but please update them too.